### ISSUE TITLE
No such index error with 2.3.8 update

### ISSUE THREAD

**Issue by LeBrad** [wallabag/wallabag#4169]
When I run make update from 2.3.x to 2.3.8, I get the following error:

>Migrating up to 20190708122957 from 20181128203230
>
> ++ migrating 20181202073750
>
>    -> DROP INDEX UNIQ_1D63E7E5C05FB297
>
>
>Migration 20181202073750 failed during Execution. Error An exception occurred while executing 'DROP INDEX UNIQ_1D63E7E5C05FB297':
>
>SQLSTATE[HY000]: General error: 1 no such index: UNIQ_1D63E7E5C05FB297

Installed via git
Wallabag 2.3.x
Debian 10 Buster
PHP 7.3.11
Dedicated hosting
SQLite 2.8.17

Maybe updating to Debian 10 broke something?

**Comment by j0k3r** [comment]
From which version are you upgrading?
That index should have been created during the initial creation.
Did you upgrade wallabag before that upgrade?

**Comment by LeBrad** [comment]
I've been updating Wallabag regularly for years. In this case, it was a working 2.3.x install that I was updating from. Maybe I can manually add that index? Also I should probably upgrade from SQLite to MariaDB since my server is not resource constrained. I know it's outside the scope of this bug, but... is there any official procedure for how to do upgrade databases? Thanks!

**Comment by j0k3r** [comment]
You can create it manually: https://github.com/wallabag/wallabag/blob/98d8e0c1b929a198da0fbcbffbc011e985b5df01/app/DoctrineMigrations/Version20160401000000.php#L49

There is no such documentation for migrating.
Depending on what you have used in wallabag, you can export articles and then re-import in the new wallabag instance.
There is no export feature for tagging rule as of 2.3.8. It'll be available for the 2.4.0

**Comment by LeBrad** [comment]
After manually creating that index, `make update` reports the error:


>An exception occurred while executing 'INSERT INTO wallabag_oauth2_clients (id, user_id, random_id, secret, redirect_uris, allowed_grant_types, name) SELECT id, user_id, random_id, secret, redirect_uris, allowed_grant_types, name FROM __temp__wallabag_oauth2_clients':
>
>SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: wallabag_oauth2_clients.name

**Comment by LeBrad** [comment]
I solved it!

I used sqlitebrowser to change `wallabag_oauth2_clients.name` to a non-null string. I then had to add several indexes that make reported missing, and update my nginx config to point to php 7.3.

Now wallabag reports that it is version 2.4.0-dev on my about page even though I was expecting it to say 2.3.8. I guess make is pulling in development channel stuff?

The other weird thing is my old clients don't show up in the API management tab even though their authorization tokens still work. Newly created clients do appear though. I guess that's a separate issue.

Anyway thanks for your help, and thanks for wallabag!
